README.md

```diff
@@ -107,3 +107,66 @@ _Args(a=2, b=<Bird.puffin: 1>, c=None)
 $ uv run examples/example_1.py --a 2 --b lark --c /somewhere/to/go
 _Args(a=2, b=<Bird.lark: 2>, c=PosixPath('/somewhere/to/go'))
 ```
+
+## Pitfall: forward-references
+
+All types used in annotations _must_ be available at runtime.
+
+Specifically, when you call `argparcel.parse(Args)`, internally it relies upon
+`typings.get_type_hints(_Args)` working. It will not work if any of the type annotations
+for fields in `Args` can't be resolved at runtime. 
+
+A plausible scenario for this to fail is when using forward references.
+
+Forward references are used when:
+- annotating with a string
+- you have `from future import __annotations__` in the module
+- you're using Python 3.14 or later
+
+In any of these cases, a linter rule like [ruff's TC003](https://docs.astral.sh/ruff/rules/typing-only-standard-library-import/) may encourage you to move an import into a `TYPE_CHECKING`-guarded block, like so:
+
+```python
+from __future__ import annotations
+
+import dataclasses
+from typing import TYPE_CHECKING
+
+import argparcel
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+
+@dataclasses.dataclass
+class Args:
+    x: Path
+
+
+argparcel.parse(Args)  # Throws NameError
```

Comment:
https://docs.python.org/3/tutorial/errors.html doesn't mention 'throw'; perhaps this would be clearer to python users less familiar with languages that throw?

```suggestion
argparcel.parse(Args)  # Raises NameError
```

Reply:
Good except!

24d3bfe